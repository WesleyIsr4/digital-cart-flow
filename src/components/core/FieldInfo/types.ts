import { ReactNode } from 'react'

export type FieldInfo = {
  children?: ReactNode
  type?: 'error' | 'info' | 'warning'
  marginTopInfo?: string
}
