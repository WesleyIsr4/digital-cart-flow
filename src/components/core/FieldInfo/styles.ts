import styled from 'styled-components'

import type * as T from './types'

const getColorByType = (type: string) => {
  switch (type) {
    case 'warning':
      return 'warningBase'
    case 'error':
      return 'negativeBase'
    default:
      return 'neutralBase'
  }
}

export const FieldInfo = styled.span<T.FieldInfo>`
  position: absolute;
  font-size: 14px;

  color: ${({ theme, type = 'info' }) => theme.colors[getColorByType(type)]};
  top: 100%;

  margin-top: ${({ marginTopInfo }) => marginTopInfo};
`
