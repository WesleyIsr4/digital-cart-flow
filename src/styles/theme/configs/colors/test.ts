import colors from './index'

describe('Colors theme', () => {
  it('Should match schema', () => {
    expect(colors).toEqual({
      scaleGray1: '#EBEBEB;',
      scaleGray2: '#DDDDDD;',
      scaleGray3: '#C9C5D4',
      scaleGray4: '#666173',
      scaleBlack: '#151516',
      primaryWhite: ' #FFFFFF',
      primaryBlue: '#191847',
      secondaryOrange: '#F5850B',
      negativeBase: '#E50000',
      warningBase: '#F9A006',
      neutralBase: '#757575',
    })
  })
})
