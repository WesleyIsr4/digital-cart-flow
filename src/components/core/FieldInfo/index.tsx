import * as S from './styles'
import type * as T from './types'

function FieldInfo({ children, type = 'info', ...rest }: T.FieldInfo) {
  if (!children) return null

  return (
    <S.FieldInfo type={type} role={type === 'error' ? 'alert' : 'info'} {...rest}>
      {children}
    </S.FieldInfo>
  )
}

export default FieldInfo
