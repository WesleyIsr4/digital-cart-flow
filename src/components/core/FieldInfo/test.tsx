import { render, screen } from 'utils/test'

import FieldInfo from './index'

describe('<FieldInfo />', () => {
  it('Should match snapshot', () => {
    const { container } = render(<FieldInfo>I am an error</FieldInfo>)
    expect(container).toMatchSnapshot()
  })

  it('should render the default error condition', () => {
    render(<FieldInfo type="error">I am an error</FieldInfo>)
    expect(screen.queryByText('I am an error')).toBeInTheDocument()
    expect(screen.queryByText('I am an error')).toHaveStyle({ color: '#E50000' })
  })

  it('should render the warning condition', () => {
    render(<FieldInfo type="warning">I am a warning</FieldInfo>)
    expect(screen.queryByText('I am a warning')).toBeInTheDocument()
    expect(screen.queryByText('I am a warning')).toHaveStyle({ color: '#F9A006' })
  })
})
