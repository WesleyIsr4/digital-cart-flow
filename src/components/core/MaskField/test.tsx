import { act } from 'react-dom/test-utils'

import { fireEvent, render, screen } from 'utils/test'

import Maskfield from './index'

describe('<Maskfield />', () => {
  it('should render the default Maskfield condition', async () => {
    const ref = () => null

    render(
      <Maskfield ref={ref} data-testid="test-maskfield" label="Sem mascara" name="sem mascara" />
    )
    const element = screen.queryByTestId('test-maskfield')

    expect(element).toBeInTheDocument()

    await act(async () => {
      if (element) await fireEvent.change(element, { target: { value: '99999999999' } })
      expect(element).toHaveValue('99999999999')
    })
  })

  it('should render the Maskfield with phone mask condition', async () => {
    const ref = () => null

    render(
      <Maskfield ref={ref} data-testid="test-maskfield" name="phone" label="Phone" mask="phone" />
    )
    const element = screen.queryByTestId('test-maskfield')

    expect(element).toBeInTheDocument()

    await act(async () => {
      if (element) await fireEvent.change(element, { target: { value: '99999999999' } })
      expect(element).toHaveValue('(99) 99999-9999')
    })
  })

  it('should render the Maskfield with date mask condition', async () => {
    const ref = () => null

    render(
      <Maskfield ref={ref} data-testid="test-maskfield" name="date" label="date" mask="date" />
    )
    const element = screen.queryByTestId('test-maskfield')

    expect(element).toBeInTheDocument()

    await act(async () => {
      if (element) await fireEvent.change(element, { target: { value: '01012020' } })
      expect(element).toHaveValue('01/01/2020')
    })
  })

  it('should render the Maskfield with monthAndYear mask condition', async () => {
    const ref = () => null

    render(
      <Maskfield
        ref={ref}
        data-testid="test-maskfield"
        name="monthAndYear"
        label="monthAndYear"
        mask="monthAndYear"
      />
    )
    const element = screen.queryByTestId('test-maskfield')

    expect(element).toBeInTheDocument()

    await act(async () => {
      if (element) await fireEvent.change(element, { target: { value: '012020' } })
      expect(element).toHaveValue('01/2020')
    })
  })
})
