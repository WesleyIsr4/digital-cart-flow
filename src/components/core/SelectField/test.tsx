// Importando React, React Testing Library e Jest
import React from 'react'

import { render, fireEvent } from 'utils/test'

import '@testing-library/jest-dom/extend-expect'
import Select from './index'

const options = [
  { value: 'option1', label: 'Opção 1' },
  { value: 'option2', label: 'Opção 2' },
  { value: 'option3', label: 'Opção 3' },
]

const mockCallback = jest.fn((selected) => selected)

describe('Select Component', () => {
  it('renders correctly', () => {
    const { getByText } = render(
      <Select
        name="teste"
        required
        options={options}
        onSelectChange={mockCallback}
        label="Selecione uma opção"
        placeholder="Selecione..."
      />
    )
    expect(getByText('Selecione uma opção')).toBeInTheDocument()
  })

  it('displays options correctly', () => {
    const { getByText } = render(
      <Select
        name="teste"
        required
        options={options}
        onSelectChange={mockCallback}
        label="Selecione uma opção"
        placeholder="Selecione..."
      />
    )
    expect(getByText('Opção 1')).toBeInTheDocument()
    expect(getByText('Opção 2')).toBeInTheDocument()
    expect(getByText('Opção 3')).toBeInTheDocument()
  })

  it('calls onSelectChange with correct value when option selected', () => {
    const { getByRole } = render(
      <Select
        required
        name="teste"
        options={options}
        onSelectChange={mockCallback}
        label="Selecione uma opção"
        placeholder="Selecione..."
      />
    )
    fireEvent.change(getByRole('combobox'), { target: { value: 'option2' } })
    expect(mockCallback).toHaveBeenCalledWith('option2')
  })
})
