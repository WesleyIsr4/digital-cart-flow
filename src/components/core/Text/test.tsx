import React from 'react'

import { render, screen } from 'utils/test'

import Text from './index'

describe('Text component', () => {
  test('renders correctly', () => {
    render(
      <Text fontSize="1rem" fontWeight={500} color="primaryBlue">
        Test
      </Text>
    )
    expect(screen.getByText('Test')).toBeInTheDocument()
  })

  test('renders with correct props', () => {
    render(
      <Text fontSize="1rem" fontWeight={500} color="scaleBlack">
        Test
      </Text>
    )
    const textElement = screen.getByText('Test')
    expect(textElement).toHaveStyle({ color: '#151516' })
  })

  test('renders the correct HTML tag', () => {
    render(
      <Text tag="h1" fontSize="1rem" fontWeight={500} color="scaleGray1">
        Test
      </Text>
    )
    expect(screen.getByRole('heading')).toBeInTheDocument()
  })

  test('renders with accessibility attributes', () => {
    render(
      <Text fontSize="1rem" fontWeight={500} color="scaleGray3" role="text" ariaLabel="Test text">
        Test
      </Text>
    )
    const textElement = screen.getByText('Test')
    expect(textElement).toHaveAttribute('role', 'text')
    expect(textElement).toHaveAttribute('aria-label', 'Test text')
  })
})
