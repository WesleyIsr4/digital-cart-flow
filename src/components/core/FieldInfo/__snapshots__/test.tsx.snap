// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldInfo /> Should match snapshot 1`] = `
.c0 {
  position: absolute;
  font-size: 14px;
  color: #757575;
  top: 100%;
}

<div>
  <span
    class="c0"
    role="info"
    type="info"
  >
    I am an error
  </span>
</div>
`;
